### ISSUE TITLE
azurerm_local_network_gateway - all address spaces are removed when modifying a single one

### ISSUE THREAD

**Issue by dribblor** [hashicorp/terraform-provider-azurerm#16327]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.4

### AzureRM Provider Version

3.0.2

### Affected Resource(s)/Data Source(s)

azurerm_local_network_gateway

### Terraform Configuration Files

```hcl
resource "azurerm_local_network_gateway" "home" {
  name                = "backHome"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  gateway_address     = "12.13.14.15"
  address_space       = [
    "10.0.0.0/16",
    "10.1.0.0/16",
    "10.2.0.0/16"
  ]
}
```

### Debug Output/Panic Output

```shell
none
```

### Expected Behaviour

Only a single address space is changed. Existing spaces are not deleted and re-added.

### Actual Behaviour

When modifying a single address space CIDR in address_space, the plan output suggests the apply will only modify a single address space. What actually happens is **all** (bar the first) address spaces are dropped, breaking connectivity with those local networks, and then they all re-added.

This happened in 2.46.0 and 2.94.0 as well.

### Steps to Reproduce

1. Create a azurerm_local_network_gateway with several different local address spaces listed.
2. Modify one of the address spaces, leaving the other address spaces unchanged.
3. Note the plan says only that element will be changed. The other address spaces in the list are unchanged/hidden.
4. Apply the plan.
5. Review the 'Create or update LocalNetworkGateway' operation in the Azure activity log. Note how it deleted all the address spaces, and then afterwards it added them all back again.

**Comment by shbali** [comment]
I had the same issue.

**Terraform Version**
1.4.6

**AzureRM Provider Version**
3.56.0

**Affected Resource(s)/Data Source(s)**
azurerm_local_network_gateway

When updating the address_space variable on the local_network_gateway resource, terraform logs shows an update accordingly, 
when in fact, on Azure, it removes all the existing addresses, adds the new address, and then and previous addresses again.

It is not an update, as Terraform shows, but rather a replacement action that causes a disruption of the local_network_gateway when the existing address_space list is momentarily removed. 

To sum up the issue. 

1. add the new address to address_space list variable
2. Terraform apply and confirm the change 
3. Delete the existing address_space list 
4. add the new address
5. add the previous address_space list from step 4 to complete the change 
6. Review the 'Create or update LocalNetworkGateway' operation in the Azure activity log.

**Comment by neilrobst** [comment]
I also experienced this same issue with provider version 4.11.0

I have just diagnosed and tested it and am submitting a PR for it...
